server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/SearchSnippetGetMethod.scala

```diff
@@ -0,0 +1,108 @@
+import org.apache.james.metrics.api.MetricFactory
+import org.reactivestreams.Publisher
+import play.api.libs.json.JsObject
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.jdk.CollectionConverters._
+import scala.util.Try
+
+class SearchSnippetGetMethod @Inject()(serializer: SearchSnippetSerializer,
+                                       searchHighlighter: SearchHighlighter,
+                                       messageIdFactory: MessageId.Factory,
+                                       val metricFactory: MetricFactory,
+                                       val sessionSupplier: SessionSupplier,
+                                       val sessionTranslator: SessionTranslator) extends MethodRequiringAccountId[SearchSnippetGetRequest] {
+
+  override val methodName: MethodName = MethodName("SearchSnippet/get")
+  override val requiredCapabilities: Set[CapabilityIdentifier] = Set(JMAP_CORE, JMAP_MAIL)
+
+  override def getRequest(mailboxSession: MailboxSession, invocation: Invocation): Either[Exception, SearchSnippetGetRequest] = serializer.deserializeSearchSnippetGetRequest(invocation.arguments.value)
+    .asEitherRequest
+
+  override def doProcess(capabilities: Set[CapabilityIdentifier], invocation: InvocationWithContext, mailboxSession: MailboxSession, request: SearchSnippetGetRequest): Publisher[InvocationWithContext] =
+    processRequest(mailboxSession, invocation.invocation, request, capabilities)
+      .map(invocationResult => InvocationWithContext(invocationResult, invocation.processingContext))
+
+  private def processRequest(mailboxSession: MailboxSession,
+                             invocation: Invocation,
+                             request: SearchSnippetGetRequest,
+                             capabilities: Set[CapabilityIdentifier]): SMono[Invocation] =
+    getSearchSnippet(request, mailboxSession, capabilities)
+      .map(searchSnippetList => SearchSnippetGetResponse.from(request.accountId, searchSnippetList.toList, request.emailIds))
+      .map(response => Invocation(
+        methodName = methodName,
+        arguments = Arguments(serializer.serialize(response).as[JsObject]),
+        methodCallId = invocation.methodCallId))
+
+  private def getSearchSnippet(request: SearchSnippetGetRequest,
+                               mailboxSession: MailboxSession,
+                               capabilities: Set[CapabilityIdentifier]): SMono[Seq[SearchSnippet]] =
+    if (request.emptyRequest) {
+      SMono.just(Seq.empty)
+    } else {
+      SMono.fromCallable(() => request.emailIds.flatMap(asMessageId(_).toOption))
+        .filter(messageIds => messageIds.nonEmpty)
+        .flatMap(messageIds => buildMultiMailboxesSearchQuery(request, messageIds, capabilities, mailboxSession)
+          .fold(e => SMono.error(e),
+            multiMailboxesSearchQuery => SFlux(searchHighlighter.highlightSearch(messageIds.asJava, multiMailboxesSearchQuery, mailboxSession))
```

Comment:
searchHighlighter do already ensure the user is doing the search on mailboxes he have access to, right?

Reply:
yes